frontend/src/metabase/browse/components/utils.tsx

```diff
@@ -0,0 +1,81 @@
+import { t } from "ttag";
+
+import { isRootCollection } from "metabase/collections/utils";
+import { getIcon } from "metabase/lib/icon";
+import * as Urls from "metabase/lib/urls";
+import type {
+  Collection,
+  CollectionItem,
+  SearchResult,
+} from "metabase-types/api";
+
+import { getCollectionName } from "../utils";
+
+export const getBreadcrumbMaxWidths = (
+  collections: Collection["effective_ancestors"],
+  totalUnitsOfWidthAvailable: number,
+  isPathEllipsified: boolean,
+) => {
+  if (!collections || collections.length < 2) {
+    return [];
+  }
+  const lengths = collections.map(
+    collection => getCollectionName(collection).length,
+  );
+  const ratio = lengths[0] / (lengths[0] + lengths[1]);
+  const firstWidth = Math.max(
+    Math.round(ratio * totalUnitsOfWidthAvailable),
+    25,
+  );
+  const secondWidth = totalUnitsOfWidthAvailable - firstWidth;
+  const padding = isPathEllipsified ? "2rem" : "1rem";
+  return [
+    `calc(${firstWidth}cqw - ${padding})`,
+    `calc(${secondWidth}cqw - ${padding})`,
+  ];
+};
+
+export const isModel = (item: CollectionItem) => item.model === "dataset";
+
+export const getModelDescription = (item: CollectionItem) => {
+  if (
+    item.collection &&
+    isRootCollection(item.collection) &&
+    isModel(item) &&
+    !item.description?.trim()
+  ) {
+    return t`A model`;
+  } else {
+    return item.description;
+  }
+};
+
+export const searchResultToCollectionItem = (
+  searchResult: SearchResult,
+): CollectionItem => {
+  const convertibleModels = [
+    "collection",
+    "dashboard",
+    "card",
+    "dataset",
+    "indexed-entity",
+    "snippet",
+  ];
+  if (!convertibleModels.includes(searchResult.model)) {
+    console.error(
+      `Cannot convert a search result with model = ${searchResult.model} to collection item. Coercing instead`,
+    );
+    return searchResult as unknown as CollectionItem;
+  }
+  return {
+    ...searchResult,
+    model: searchResult.model as CollectionItem["model"],
+    id: searchResult.id as number,
+    getIcon: () => getIcon(searchResult),
+    collection_id: searchResult.collection.id,
+    collection: searchResult.collection as Collection,
+    getUrl: () => Urls.modelToUrl(searchResult) as string,
+    moderated_status: searchResult.moderated_status ?? undefined,
+    can_write: searchResult.can_write ?? undefined,
+  };
+};
```

Comment:
This is a red flag for me. I understand why you did it, but I do wonder if the simpler approach here might have been to not try to make search items fit into the table designed for collection items, but to do something a bit more straightforward.